### ISSUE TITLE
Error backtrace should print names of operations/attributes/... in which the error occurred

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#1196]
Consider the following error backtrace:
```gap-repl
gap> s:=Semigroup([[0,1],[0,0]]*Z(2));;
gap> IsomorphismTransformationMonoid(s);
Error, IsomorphismTransformationMonoid: usage,
the argument must be a semigroup with a multiplicative neutral element, at GAPROOT/lib/semitran.gi:360 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:2
```
The message `<function "unknown">( <arguments> )` is not very helpful. Contrast with this:
```gap-repl
gap> Group("foo");
Error, usage: Group(<gen>,...), Group(<gens>), Group(<gens>,<id>) at GAPROOT/lib/grp.gi:4410 called from
<function "Group">( <arguments> )
 called from read-eval loop at *stdin*:1
```
which still is not super informative, but at least contains the name of the calling function. The difference seems to be caused by the fact that `IsomorphismTransformationMonoid` is an attribute, and method functions don't have a name ?!?

**Comment by markuspf** [comment]
If you just call 

```InstallMethod( [...], function(a) return "Hello"; end)```

what name should that function have? I think most usefully one could display which method was selected, and possibly some location info (where the InstallMethod is in this case).

I am sure I have some not-very-mature hacks around to do that.

**Comment by fingolfin** [comment]
Let me clarify that this is really about what the backtrace display -- which for me is *not* necessarily the same as the "name" of the function. I think in backtraces, to find the actual code in which an error occurs, of course the `FILE:LINE` pair is the most important bit. The name of the function or operation to m serves another purpose: It helps me locate where in my code I messed up.

To stay with the original example of this issue: If I wrote a program which runs into an error, then the information "error occurred because bad args were passed to `IsomorphismTransformationMonoid`" can be immediately helpful, while the information that `Error` was called in `GAPROOT/lib/semitran.gi:360` is not -- it is only after I open the file and navigate to that line, and in the process find out which operation was called. (Of course for me personally, that's usually just a single Cmd-click away, thanks to iTerm2 being awesome; but if I debug code via SSH or one somebody else's computer, I loose that).

BTW, I also think that `<function "Group">( <arguments> )` is not very helpful -- can't we just print something like `Group(...)` ?

To illustrate what I mean, here is how IMHO the two errors above could look to be more helpful:
```
gap> IsomorphismTransformationMonoid(s);
Error, usage: the argument must be a semigroup with a multiplicative neutral element
in method for attribute IsomorphismTransformationMonoid(...) at GAPROOT/lib/semitran.gi:360
called from read-eval loop at *stdin*:2
```
and
```
Error, usage: Group(<gen>,...), Group(<gens>), Group(<gens>,<id>)
in global function Group() at GAPROOT/lib/grp.gi:4410
called from read-eval loop at *stdin*:1
```

**Comment by fingolfin** [comment]
Looking closer at this, I think I also find the backtrace in general somewhat confusing (perhaps this should be a separate issue). To explain, I created a GAP file with this content:
```
foo:=function()
  Error("Something went wrong");
end;
bar:=function()
  foo();
end;
quux:=function()
  bar();
end;

foo2:=function()
  return 1/0;
end;
bar2:=function()
  foo2();
end;
quux2:=function()
  bar2();
end;
```
After reading this, I observe these error messages:
```
gap> quux();
Error, Something went wrong at err-test.g:2 called from
foo(  ); at err-test.g:5 called from
bar(  ); at err-test.g:8 called from
<function "quux">( <arguments> )
 called from read-eval loop at *stdin*:1
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk>
```
Note how "foo(  ); in the second error line refers to the fact that `err-test.g:5` (which is inside function `bar`) contains the statement `foo();`. Yet on can easily misinterpret the first two lines as saying "called from function foo". Esp. since in the fourth line, we really mean "called from function quux" -- of course it is displayed completely differently, but nevertheless I find it confusing. I think the term "called from" is partially responsible for that. Also note how the line "called from read-eval loop at *stdin*:1" is subtly different from the other again -- instead of ending in "called from", it starts with it.

Another source of my confusing is that it says `foo(  )`, even though the actual code says `foo()` -- of course this is because we don't store the actual text of the file. (Perhaps we should change that?)

Also, if a function invokes `Error()`, things are different from when it triggers an error; the latter looks like this, which I find slightly less confusing (at least the start), because the fact that code line `return 1 / 0;` is shown at least gives a clue that `foo2(  )` below might also be a code line.
```
gap> quux2();
Error, Rational operations: <divisor> must not be zero in
  return 1 / 0; at err-test.g:12 called from
foo2(  ); at err-test.g:15 called from
bar2(  ); at err-test.g:18 called from
<function "quux2">( <arguments> )
 called from read-eval loop at *stdin*:2
you can replace <divisor> via 'return <divisor>;'
brk>
```

**Comment by fingolfin** [comment]
See also #917 and #358.

**Comment by fingolfin** [comment]
This is fixed now thanks to #6257, #6263 and a few more
